inmoose/utils/lm.py

```diff
@@ -0,0 +1,141 @@
+# -----------------------------------------------------------------------------
+# Copyright (C) 2024 Maximilien Colange
+
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+# -----------------------------------------------------------------------------
+
+import numpy as np
+from statsmodels.regression.linear_model import WLS
+
+
+class lm_wfit:
```

Comment:
I don't know about best practice, but why create 2 different classes here including one that derive from the other when the only difference seems to be the weights to use in the least-square-error fit? Can't we use `lm_wfit` with a default weight of 1 for the `lm_fit` class directly in the code?

Reply:
The difference between using a default weight and using 2 classes is in terms of API.
In both cases, the regular fit is implemented as a special case of the weighted fit.
Yet weighted fit is more specialized (in terms of use cases) and most users would be satisfied with the regular fit.

Exposing two separate routines is in fact the approach taken in R (functions `lm.fit` and `lm.wfit`), which we try to mimic here, and in `scipy` (classes `OLS` and `WLS`), even though the regular fit is implemented as a special case of the weighted fit.